nevergrad/functions/stsp/core.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Based on a discussion at Dagstuhl's seminar on Computational Intelligence in Games with:
+# - Dan Ashlock
+# - Chiara Sironi
+# - Guenter Rudolph
+# - Jialin Liu
+
+from math import pi, cos, sin
+from typing import Any
+from typing import List
+import matplotlib.pyplot as plt
+import numpy as np
+from ... import instrumentation as inst
+from ...instrumentation.multivariables import Instrumentation
+
+
+# pylint: disable=too-many-instance-attributes,too-many-arguments,too-many-statements,too-many-locals
+class STSP(inst.InstrumentedFunction):
+
+    def __init__(self, seed: int = 0, the_dimension: int = 500) -> None:
+        instrumentation = Instrumentation(inst.var.Array(the_dimension))
+        self.x = instrumentation.random_state.normal(size=the_dimension)
+        self.y = instrumentation.random_state.normal(size=the_dimension)
+        super().__init__(self._simulate_stsp, instrumentation)
+        self._descriptors.update(seed=seed)
+
+    def _simulate_stsp(self, x: np.ndarray) -> float:
+        order = np.argsort(x)
+        self.order = order
```

Comment:
See other PR (they should have been decoupled)
you should not set attributes which were not created in the init